pkg/controller/drain/drain_controller.go

```diff
@@ -359,13 +393,47 @@ func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) erro
 		ctrl.logNode(node, "cordoning")
 		// perform cordon
 		if err := ctrl.cordonOrUncordonNode(true, node, drainer); err != nil {
+			Nerr := upgrademonitor.GenerateAndApplyMachineConfigNodes(&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateExecuted, Reason: string(v1alpha1.MachineConfigNodeUpdateCordoned), Message: fmt.Sprintf("Failed to Cordon Node as part of In progress update phase")},
+				&upgrademonitor.Condition{State: v1alpha1.MachineConfigNodeUpdateCordoned, Reason: fmt.Sprintf("%s%s", string(v1alpha1.MachineConfigNodeUpdateExecuted), string(v1alpha1.MachineConfigNodeUpdateCordoned)), Message: fmt.Sprintf("Error: Failed to Cordon node. Error is %s, The node is reporting Unschedulable = %t", err.Error(), node.Spec.Unschedulable)},
+				metav1.ConditionUnknown,
+				metav1.ConditionUnknown,
+				node,
+				ctrl.client,
+				ctrl.featureGatesAccessor,
+			)
+			if Nerr != nil {
+				klog.Errorf("Error making MCN for Cordon Failure: %w", Nerr)
+			}
 			return fmt.Errorf("node %s: failed to cordon: %w", node.Name, err)
```

Comment:
question (non-blocking): Would it make sense to aggregate the error from `GenerateAndApplyMachineConfigNodes()` and the `cordonOrUncordonNode()` error before returning them whenever `GenerateAndApplyMachineConfigNodes()` returns an error?

In other words:
```golang
// aggerror is https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go

cordonOrUncordonErr := fmt.Errorf("node %s: failed to cordon: %w", node.Name, err)

if Nerr != nil {
    wrappedNerr := fmt.Errorf("error making MCN for Cordon Failure: %w", Nerr)
    klog.Errorf(wrappedNerr)  
    return aggerror.NewAggregate([]error{wrappedNerr, cordonOrUncordonErr})
}

return cordonErr
```

(If just emitting the error from `GenerateAndApplyMachineConfigNodes()` is enough

Reply:
hmmm this might have some unintended consequences in corner cases. I am going to table this and attach it to the general concerns over cordon/uncordon in 4.16.